dpnp/dpnp_iface_mathematical.py

```diff
@@ -3255,14 +3264,31 @@ def interp(x, xp, fp, left=None, right=None, period=None):
 
 """
 
-maximum = DPNPBinaryFunc(
+_maximum_impl = DPNPBinaryFunc(
     "maximum",
     ti._maximum_result_type,
     ti._maximum,
     _MAXIMUM_DOCSTRING,
 )
 
 
+@wraps(_maximum_impl)
+def maximum(*args, **kwargs):
+    """
+    Wrapper around `_maximum_impl` that emits a DeprecationWarning
+    when `out` is passed positionally.
+
+    """
+    if len(args) >= 3 and "out" not in kwargs:
```

Comment:
Why do we need to check that? What is the use case?
```suggestion
    if len(args) >= 3:
```

Reply:
It'd make sense to use `nin`:
```suggestion
    if len(args) > _maximum_impl.nin:
```